docs/docs/llms/responses-agent-intro/index.mdx

```diff
@@ -0,0 +1,310 @@
+- Support for multiple messages and intermediate steps
+- Support for multi-agent scenarios
+- Ensure compatibility with MLflow logging, tracing, and model serving
+
+We recommend `ResponsesAgent` instead of <APILink fn="mlflow.pyfunc.ChatModel">`ChatModel`</APILink> and <APILink fn="mlflow.pyfunc.ChatAgent">`ChatAgent`</APILink>, as it has all the benefits of `ChatAgent` and is compatible with OpenAIs Responses API.
+
+
+## Author a ResponsesAgent
+
+To author a ResponsesAgent, create a subclass of the ResponsesAgent and implement a predict method using any agent authoring framework. You must be using `pydantic>=2` to use ResponsesAgent.
+
+When creating output objects, you should only create `ResponsesAgentResponse` and `ResponsesAgentStreamEvent` pydantic objects. The other classes in `mlflow.types.responses_helpers` are for validation only. Below is an example of a tool calling sequence:
+```python
+from mlflow.types.responses import ResponsesAgentResponse
+
+ResponsesAgentResponse(
+    output=[
+        {
+            "arguments": '{"code":"result = 4 * 3\\nprint(result)"}',
+            "call_id": "call_1",
+            "name": "python_exec",
+            "type": "function_call",
+            "id": "fc_1",
+            "status": "completed",
+        },
+        {
+            "type": "function_call_output",
+            "call_id": "call_1",
+            "output": "12\n",
+        },
+        {
+            "id": "msg_1",
+            "content": [
+                {
+                    "annotations": [],
+                    "text": "The result of 4 * 3 in Python is 12.",
+                    "type": "output_text",
+                }
+            ],
+            "role": "assistant",
+            "status": "completed",
+            "type": "message",
+        },
+    ]
+)
+```
+An equivalent stream would look like:
+```python
+from mlflow.types.responses import ResponsesAgentStreamEvent
+
+ResponsesAgentStreamEvent(
+    type="response.output_item.done",
+    item={
+        "arguments": '{"code":"result = 4 * 3\\nprint(result)"}',
+        "call_id": "call_1",
+        "name": "python_exec",
+        "type": "function_call",
+        "id": "fc_1",
+        "status": "completed",
+    },
+)
+ResponsesAgentStreamEvent(
+    type="response.output_item.done",
+    item={
+        "type": "function_call_output",
+        "call_id": "call_1",
+        "output": "12\n",
+    },
+)
+ResponsesAgentStreamEvent(
+    type="response.output_item.done",
+    item={
+        "id": "msg_1",
+        "content": [
+            {
+                "annotations": [],
+                "text": "The result of 4 * 3 in Python is 12.",
+                "type": "output_text",
+            }
+        ],
+        "role": "assistant",
+        "status": "completed",
+        "type": "message",
+    },
+)
+```
+
+
+### Streaming a ResponsesAgent
+
+
+See the [example below](#example-responsesagent) for a full implementation!
+
```

Comment:
This seems like maybe the right place to mention the details about streaming classes etc